src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -474,6 +473,16 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator, IRBlobSe
         private async Task RunWorker(CancellationToken ct) {
             TaskUtilities.AssertIsOnBackgroundThread();
 
+            // Spin until the worker task is registered.
+            while (_runTask == null) {
+                await Task.Yield();
```

Comment:
Looks like a hack.
Probably we replace these 3 tasks with 3 `TaskCompletionSourceEx` instances and cancel them in `finally` block